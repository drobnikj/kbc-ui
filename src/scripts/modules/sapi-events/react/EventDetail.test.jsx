import React from 'react';
import EventDetail from './EventDetail';
import { fromJS } from 'immutable';


const link = {
  to: 'something'
};

const event = fromJS({
  'id': 612801925,
  'event': 'ext.docker.',
  'component': 'docker',
  'message': 'Running Component ex-generic-v2',
  'description': '',
  'type': 'info',
  'runId': '320118552',
  'created': '2017-09-25T19:59:02+0200',
  'configurationId': null,
  'objectId': '',
  'objectName': '',
  'objectType': '',
  'context': {
    'remoteAddr': '52.206.109.126',
    'httpReferer': null,
    'httpUserAgent': 'Keboola Storage API PHP Client\/7.0.2 docker',
    'apiVersion': 'v2'
  },
  'params': {},
  'results': {},
  'performance': {},
  'token': {
    'id': 3810,
    'name': 'martin@keboola.com'
  },
  'uri': 'https:\/\/connection.keboola.com\/v2\/storage\/events\/612801925',
  'attachments': []
});

describe('<EventDetail />', function() {
  it('should render event', function() {
    shallowSnapshot(
      <EventDetail
        event={event}
        link={link}
      />
    );
  });
});
