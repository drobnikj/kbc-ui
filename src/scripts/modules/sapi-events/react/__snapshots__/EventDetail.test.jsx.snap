exports[`<EventDetail /> should render event 1`] = `
<div>
  <Link
    activeClassName="active"
    className=""
    to="something">
    <span
      className="fa fa-chevron-left" />
     Back
  </Link>
  <h2>
    Event 
    612801925
  </h2>
  <div
    className="well">
    <NewLineToBr
      text="Running Component ex-generic-v2" />
  </div>
  <div
    className="row">
    <div
      className="col-md-3">
      Created
    </div>
    <div
      className="col-md-9">
      2017-09-25 17:59:02
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-3">
      Component
    </div>
    <div
      className="col-md-9">
      docker
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-3">
      Configuration ID
    </div>
    <div
      className="col-md-9">
      N/A
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-3">
      Run ID
    </div>
    <div
      className="col-md-9">
      320118552
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-3">
      Creator
    </div>
    <div
      className="col-md-9">
      martin@keboola.com
    </div>
  </div>
  <div>
    <h3>
      Context
    </h3>
    <Tree
      data={
        Object {
          "apiVersion": "v2",
          "httpReferer": null,
          "httpUserAgent": "Keboola Storage API PHP Client/7.0.2 docker",
          "remoteAddr": "52.206.109.126",
        }
      } />
  </div>
</div>
`;
